src/gossip/table.zig

```diff
@@ -398,22 +399,23 @@ pub const GossipTable = struct {
 
     /// Since a node may be represented with ContactInfo or LegacyContactInfo,
     /// this function checks for both, and efficiently returns the data as
-    /// ContactInfo, regardless of how it was received.
-    pub fn getContactInfo(self: *const Self, pubkey: Pubkey) ?ContactInfo {
+    /// ThreadSafeContactInfo, regardless of how it was received.
+    pub fn getThreadSafeContactInfo(self: *const Self, pubkey: Pubkey) ?ThreadSafeContactInfo {
         const label = GossipKey{ .ContactInfo = pubkey };
         if (self.store.get(label)) |v| {
-            return v.value.data.ContactInfo;
+            return ThreadSafeContactInfo.fromContactInfo(v.value.data.ContactInfo);
         } else {
-            return self.converted_contact_infos.get(pubkey);
+            return ThreadSafeContactInfo.fromContactInfo(self.converted_contact_infos.get(pubkey) orelse return null);
         }
     }
 
-    pub fn genericGetWithCursor(
+    pub fn genericGetEntriesWithCursor(
+        allocator: ?std.mem.Allocator,
```

Comment:
In all of the usages where null is passed here, the return contains a pointer to data owned by the table.

- getVotesWithCursor
- getEpochSlotsWithCursor
- getDuplicateShredsWithCursor

So it would be unsafe to use the data returned by those methods across a lock boundary.

Those methods are not actually used anywhere. My first instinct is to remove them. Not sure why we have unused methods. I'm guessing it's for feature parity with the agave implementation. In that case, I would probably just change them to accept an allocator. It seems preferable to default to safety and require close inspection to be fast, rather than defaulting to speed and requiring close inspection to be safe.

Also I would probably make this method private and document that it may be unsafe to use the return value across lock boundaries if the allocator is not set.